bindinfo/cache.go

```diff
@@ -191,3 +228,68 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (h *Handle) AddGlobalBind(normalizedOriginSQL, bindSQL, defaultDB, charset, collation string) error {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	ctx := context.TODO() //we need a new ctx to execute a transcation
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	_, err := exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			_, err = exec.Execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := exec.Execute(ctx, "ROLLBACK")
+			terror.Log(rbErr)
+		}
+	}()
+
+	sql := fmt.Sprintf("DELETE FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		normalizedOriginSQL, defaultDB)
+	_, err = exec.Execute(ctx, sql)
+	if err != nil {
+		return err
+	}
+	txn, err := h.ctx.Txn(true) //active a new txn to get startTs as bind's createTs.
```

Comment:
The comment is incorrect. This txn is actived when we execute the `BEGIN` sql in line 239. The only purpose of calling this function is to get the txn start ts.